Fix #1280: mention preferred doc comment syntax and update links.

mehaase opened the pull request:
At Bob Nystrom's suggestion, instead of referencing Completer as a
good example of doc comments (it uses the /*\* syntax), reference
`path` instead, which is outside of `dart:core`.

For background, see: https://groups.google.com/a/dartlang.org/forum/#!topic/misc/KHScJ4nfL2s

mehaase committed:
Fix #1280: mention preferred doc comment syntax and update links.

At Bob Nystrom's suggestion, instead of referencing Completer as a
good example of doc comments (it uses the /** syntax), reference
`path` instead, which is outside of `dart:core`.

mehaase commented:
I have signed the CLA.

kwalrath commented on src/site/articles/doc-comment-guidelines/index.markdown:
```diff
@@ -58,9 +64,9 @@ And one for a variable or property:
-[source code](http://code.google.com/p/dart/source/browse/branches/bleeding_edge/dart/sdk/lib/async/future.dart)).
+For an example of a library with doc comments, see the `path` library.
+([generated doc](http://www.dartdocs.org/documentation/path/1.3.3/index.html),
+[source code](https://github.com/dart-lang/path/blob/1.3.3/lib/path.dart)).
```
It looks like the source for the linked-to page is actually this (which surprised me): https://raw.githubusercontent.com/dart-lang/path/1.3.3/README.md

kwalrath commented on src/site/articles/doc-comment-guidelines/index.markdown:
```diff
@@ -58,9 +64,9 @@ And one for a variable or property:
-([generated doc](http://api.dartlang.org/dart_async/Completer.html),
-[source code](http://code.google.com/p/dart/source/browse/branches/bleeding_edge/dart/sdk/lib/async/future.dart)).
+For an example of a library with doc comments, see the `path` library.
+([generated doc](http://www.dartdocs.org/documentation/path/1.3.3/index.html),
```
Change to http://www.dartdocs.org/documentation/path/1.3.3/index.html#path/path

kwalrath commented on src/site/articles/doc-comment-guidelines/index.markdown:
```diff
@@ -36,6 +36,12 @@ See also these related articles:
 
 ## Examples
 
+Dart supports two syntaxes for doc comments (`///` and `/**`), but we
```
I'd rather not explanation the reasons for preferring /// over /** here—the style guide seems like the place for that. Maybe we could do something like this:

```
Here's a typical comment for a function or method:
...
<aside class="alert alert-info" markdown="1">
<b>Style note:</b>
Dart supports two syntaxes for doc comments: `///` and `/**`.
This document follows the
[style guide recommendation](/articles/style-guide/#do-use-doc-comments-when-commenting-members-and-types)
of using `///`. Older source code tends to use `/**`.
</aside>
```

Then we could add the justification to the style guide. (You could do that in this pull request, or I can do it separately.)

kwalrath commented:
@mehaase are you going to be able to make these changes? If not, I can do it. Just let me know...

mehaase commented:
I was planning too but I'm mired in some Angular.dart bugs right now and I haven't had any time to spend on this. I can probably update the PR tomorrow, but I also won't be offended if you take over from here.

kwalrath commented:
Either way is fine. If you haven't done it by, let's say, Friday, then I'll handle it.

mehaase committed:
Fix #1280: incorporate feedback from Kathy Walrath

Move the justification for /// to the style guide, but add a note
and link in the doc comment guidelines so users can find it. Fix
the URL for the generated `path` documentation.

mehaase commented:
Kathy, please take a look at my last commit.

kwalrath commented:
LGTM. Thanks!
